plugin/pkg/auth/authorizer/rbac/rbac.go

```diff
@@ -74,6 +82,21 @@ func (r *RBACAuthorizer) Authorize(requestAttributes authorizer.Attributes) (aut
 
 	r.authorizationRuleResolver.VisitRulesFor(requestAttributes.GetUser(), requestAttributes.GetNamespace(), ruleCheckingVisitor.visit)
 	if ruleCheckingVisitor.allowed {
+		ae := requestAttributes.GetAuditEvent()
```

Comment:
getting an object out of requestAttributes in order to write back to it is a fundamental change in the way the authorizer attributes are used, and breaks remote delegation cases for that interface

Reply:
See above. We need an explicit data structure here. Not a huge event with unclear ownership.